spec.html

```diff
@@ -4393,6 +4393,20 @@ <h1>The Lexical Environment and Environment Record Specification Types</h1>
       <p>The Lexical Environment and Environment Record types are used to explain the behaviour of name resolution in nested functions and blocks. These types and the operations upon them are defined in <emu-xref href="#sec-lexical-environments"></emu-xref>.</p>
     </emu-clause>
 
+    <emu-clause id="sec-abstract-closure">
+      <h1>The Abstract Closure Specification Type</h1>
+      <p>The <dfn>abstract closure</dfn> specification type is used to refer to algorithmic steps together with a collection of values. Abstract closures are meta-values and are invoked using function application style such as _closure_(_arg1_, _arg2_). Like abstract operations, invocations perform the algorithmic steps described by the abstract closure. Unlike abstract operations, abstract closures are meta-values and must be created.</p>
+      <p>In algorithmic steps that create an abstract closure, values are captured with the verb "capture" followed by a list of aliases. The value associated with each alias at that time is captured. In algorithmic steps describing calling an abstract closure, captured values are referred to by the alias that was used to capture the value.</p>
+      <p>Abstract closures are created inline as part of other algorithms, shown in the following example.</p>
+      <emu-alg>
+        1. Let _addend_ be *41*.
+        1. Let _closure_ be a new abstract closure with parameters (_x_) that captures _addend_ and performs the following steps when called:
+          1. Return _x_ + _addend_.
```

Comment:
What exactly does this return statement do? I would have assumed that this is subject to [implicit Completion values](https://tc39.es/ecma262/#sec-implicit-completion-values), but the line below (Let _val_ be _closure_(1)) doesn't use ReturnIfAbrupt or any of its shorthands.

Reply:
If you believe that the closure returns a completion record, then `Let _val_ be _closure_(1)` doesn't conflict with that. In that view, `_val_` is 'assigned' a completion record (and in the subsequent Assert, it is implicitly unpacked to the `[[Value]]` field of that record).

Reply:
Aha I missed this comment in the spec:

> Any reference to a Completion Record value that is in a context that does not explicitly require a complete Completion Record value is equivalent to an explicit reference to the [[Value]] field of the Completion Record value unless the Completion Record is an abrupt completion.

Still good to have clarification though.

Reply:
Clarified that if an abstract closure returns a Completion Record, it must be either normal or throw. @bakkot noticed that Matchers already don't return completion records.